tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1226,6 +1226,10 @@ void EnqueueRecordEventCommand::process() {
             sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd) + dispatch_constants::EVENT_PADDED_SIZE,
             PCIE_ALIGNMENT);  // CQ_PREFETCH_CMD_RELAY_INLINE + CQ_DISPATCH_CMD_WRITE_LINEAR_HOST + event ID
 
+    if (not device->is_mmio_capable()) {
```

Comment:
Will add a comment, but this is because for the non-mmio chip record_event, dispatch_d gets the `dispatch_write_remote` command, which notifies prefetcher of event completion.

Reply:
Comment added.